File: cpp-test-files/09.cpp
//Öröklődés
#include <iostream>
#include <string>
using namespace std;

class City {
protected:
    string name;
    int population;

public:
    City(const string& name, int population)
        : name(name), population(population) {}

    void displayInfo() const {
        cout << "City: " << name << endl;
        cout << "Population: " << population << endl;
    }
};

class CapitalCity : public City {
private:
    string country;

public:
    CapitalCity(const string& name, int population, const string& country)
        : City(name, population), country(country) {}

    void displayInfo() const {
        City::displayInfo();
        cout << "Country: " << country << endl;
    }
};

File: java-files/09.java
public class City {
    protected String name;
    protected int population;

    public City(String name, int population) {
        this.name = name;
        this.population = population;
    }

    public void displayInfo() {
        this.displayInfo();
        System.out.println("City: " + name + ", Population: " + population);
    }
}

public class CapitalCity extends City {
    private String country;

    public CapitalCity(String name, int population, String country) {
        this.name = name;
        this.population = population;
        this.country = country;
    }

    public void displayInfo() {
        this.displayInfo();
        System.out.println("Country: " + country);
    }
}